Core/OxidEE_Events.php

```diff
@@ -200,46 +197,66 @@ private static function _deleteOrderTransactionTable()
     {
         $sQuery = "DROP TABLE IF EXISTS `wdoxidee_ordertransactions`";
 
-        $oDb = oxDb::getDb();
-        $oDb->Execute($sQuery);
+        self::$oDb->Execute($sQuery);
+    }
+
+    /**
+     * Add Wirecard's payment methods defined in payments.xml
+     */
+    private static function _addPaymentMethods()
+    {
+        $oLogger = Registry::getLogger();
+        $oConfig = Registry::getConfig();
+        $sShopBaseURL = $oConfig->getShopUrl();
+        $xmldata = simplexml_load_file($sShopBaseURL . "modules/wirecard/paymentgateway/payments.xml");
+        if ($xmldata) {
+            foreach ($xmldata->payment as $payment) {
+                $array = array(
+                    "OXID" => $payment->oxid
+                );
+
+                $sQuery = "INSERT INTO " . 'oxpayments' . "(`OXID`, `OXACTIVE`, `OXTOAMOUNT`, `OXDESC`, `OXDESC_1`,
+                 `WDOXIDEE_LOGO`, `WDOXIDEE_TRANSACTIONTYPE`, `WDOXIDEE_APIURL`, `WDOXIDEE_MAID`, `WDOXIDEE_SECRET`,
+                 `WDOXIDEE_HTTPUSER`, `WDOXIDEE_HTTPPASS`, `WDOXIDEE_ISWIRECARD`, `WDOXIDEE_BASKET`,
+                 `WDOXIDEE_DESCRIPTOR`, `WDOXIDEE_ADDITIONAL_INFO`) VALUES ('{$payment->oxid}', '{$payment->oxactive}',
+                 '{$payment->oxtoamount}', '{$payment->oxdesc}', '{$payment->oxdesc_1}', '{$payment->wdoxidee_logo}',
+                 '{$payment->wdoxidee_transactiontype}', '{$payment->wdoxidee_apiurl}', '{$payment->wdoxidee_maid}',
+                 '{$payment->wdoxidee_secret}', '{$payment->wdoxidee_httpuser}', '{$payment->wdoxidee_httppass}',
+                 '{$payment->wdoxidee_iswirecard}', '{$payment->wdoxidee_basket}', '{$payment->wdoxidee_descriptor}',
+                 '{$payment->wdoxidee_additional_info}');";
+                self::_insertRowIfNotExists('oxpayments', $array, $sQuery);
+
+                $sRandomOxidId = substr(str_shuffle(md5(time())), 0, 15);
+                self::_insertRowIfNotExists(
+                    'oxobject2payment',
+                    array('OXPAYMENTID' => $payment->oxid),
+                    "INSERT INTO oxobject2payment (`OXID`, `OXPAYMENTID`, `OXOBJECTID`, `OXTYPE`) VALUES
+                ('{$sRandomOxidId}', '{$payment->oxid}', 'oxidstandard', 'oxdelset');"
```

Comment:
Please, either put both braces in separate lines or none in order to make it consistent.